more changes for the extension.

szjfhb opened the pull request:
1. all new feature changes for extension.
2. remove content-scraping.

Thanks.

Commit:
1. add storage to sync data.

2. add demo page. modify content_script match rule (current only for debug)
3. content send cart status to backendjs & save storage for sync data.
4. backendjs will show cart items number & adjust extesion popup page according to it's status.
By Odin

Commit:
unset changes.

By Odin

Commit:
append code for get impact from ghgi api.

By Odin

Commit:
change some workflow, fix some cartitem read bugs, correct some unit convert case and so on.

By Odin

Commit:
adjust some for demo page.  popup page show more info in console.

By Odin

Commit:
modify demo page for show bage related function.

By Odin

Commit:
1.apply carbon cost fee rate, and show related carbon fee.

2. in demo page, reload storage data when refresh page
By Odin

Commit:
modify demo for auto open when install//update. easy for debug.

By Odin

Commit:
1. remove script for enable content js for demo page. Demo now run inside extension.

2. adjust UI for empty/no-food, share one page.
By Odin

Commit:
missing common.js for popup.js....  But, think about seperate the js for different Popup pages...

By Odin Xiong

Commit:
add poppins font.

By Odin Xiong

Commit:
set badge text before get ghgi. only set popup status in ghgi request.

By Odin

Commit:
adjust codes for logs. TODO: we need add extension description text.

By Odin

Commit:
1. fix sometimes UI error. due to not-empty not exist any more.

2. modify fee rate to 0.11 per lb,  as $50 for 1000kg.
3. some other mirror changes.
By Odin

Commit:
adjust UI for offset.html & empty.html.

By Odin

Commit:
1. change convert cost rate, 2. catch error, not check field.  3.some other small optimize.

By Odin

Commit:
add error handle and error page. adjust some logic of zero check and so on.

By Odin

Commit:
adjust UI.

By Odin

Commit:
adjust UI.

By Odin

Commit:
adjust UI.

By Odin

Commit:
adjust logic for read data from instacart.

By Odin

Commit:
handle logic for 24h data expired.

By Odin

Commit:
handle logic for outdated badge color & page change.  Also error page append retry function.

By Odin

Commit:
1. correct typo of container...

2. when in homepage, click cart button can't read cart info. need choose store first.
3. add delay for load Carbon Impact data. Also handle the related texts.
By Odin

Commit:
1. change match_conf to 0.2

2. change the calc delay to 1s.
3. cart UI logic optimize. reduce wait to read cart, avoid trigger multiple times to post data, also set outdated when read, user left cart dialog.
By Odin

Commit:
remove content-scraping. that's not needed anymore.

By Odin

Commit:
add readme.

By Odin

szjfhb committed:
Update readme.MD

szjfhb committed:
Update readme.MD

Commit:
update readme, and setup a pack batch in windows.

By Odin

szjfhb committed:
Update readme.MD

append the pack related info. some readme optimize.

Commit:
update part about pack the extension.

By Odin

Commit:
some optimize for demo page.

By Odin

Commit:
set not open demo page. only when get message to open debug.

By Odin

Commit:
append display area for show current cart status.

By Odin

Commit:
update readme.

By Odin

tonychen2 commented on web-plugin/popup/Outdated.html:
```diff
@@ -0,0 +1,25 @@
+		</div>
+		<div class="container">
+			<!--insert make cart carbon neutral image-->
+			<img src="../img/Sexy_Tofu_2.png" style="height: 72px;" />
```
```suggestion
			<img src="../img/error-tofu.png" style="height: 72px;" />
```

tonychen2 commented on web-plugin/popup/error.html:
```diff
@@ -0,0 +1,26 @@
+		</div>
+		<div class="container">
+			<!--insert make cart carbon neutral image-->
+			<img src="../img/Sexy_Tofu_2.png" style="height: 72px;" />
```
```suggestion
			<img src="../img/error-tofu.png" style="height: 72px;" />
```

tonychen2 commented on web-plugin/readme.MD:
```diff
@@ -0,0 +1,212 @@
+│      48x48.png
+│      close.svg
+│      gray-filler-img.png
+│      Sexy_Tofu_2.png
```
```suggestion
│      error-tofu.png
```

tonychen2 commented on web-plugin/popup/offset.html:
```diff
@@ -0,0 +1,49 @@
+<body>
+	<div class="BoardContainer">
+		<div class="txtbubble">
+			<img src="../img/Sexy_tofu_dither.png" class="tofoIcon" />
```
```suggestion
			<img src="../img/dither-tofu.png" class="tofuIcon" />
```

tonychen2 commented on web-plugin/popup/offset.css:
```diff
@@ -0,0 +1,157 @@
+  flex-grow: 0;
+}
+
+.tofoIcon {
```
```suggestion
.tofuIcon {
```

tonychen2 commented on web-plugin/readme.MD:
```diff
@@ -0,0 +1,212 @@
+│      close.svg
+│      gray-filler-img.png
+│      Sexy_Tofu_2.png
+│      Sexy_tofu_dither.png
```
```suggestion
│      dither-tofu.png
```

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,268 @@
+}
+
+const GHGI_API_ADDRESS = 'https://api.sexytofu.org/api.ghgi.org:443';
+const GHGI_CONFIDENCE_LIMIT = 0.2; //need try found a reasonable limit. 0.5 will ignore too much, like organic banana.
```
建议咱们放到一个config file里？

szjfhb replied:
en en, 的确可以。 我看看能不能放到 插件配置里。

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,268 @@
+
+const GHGI_API_ADDRESS = 'https://api.sexytofu.org/api.ghgi.org:443';
+const GHGI_CONFIDENCE_LIMIT = 0.2; //need try found a reasonable limit. 0.5 will ignore too much, like organic banana.
+const G_TO_POUND = 0.00220462262185;
```
这种我觉得就留在background.js里或者单独建一个constant file都行

tonychen2 commented on web-plugin/js/common.js:
```diff
@@ -0,0 +1,24 @@
+class TofuItem {
```
这个文件是干嘛的呀？

szjfhb replied:
搞了个 class，传输后失效了。 ：） 用法可读性更好。 在 content 里有引用

tonychen2 replied:
什么叫“传输后失效了”？

我只是不太明白TofuItem为啥会和另外那俩constants在同一个文件里哈

szjfhb replied:
哦, TofuItem, 本来是好玩建的 class,  js 里 class 好像并不好用.
这个 common.js 是想用来存储这些公用的常量之类的... :) 它在这里有些怪.  可以考虑移出来...

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,268 @@
+
+chrome.runtime.onInstalled.addListener(details => {
+    if (details.reason === chrome.runtime.OnInstalledReason.INSTALL) {
+        //TODO: here we need a page outside of the extension to collect feedback.
```
安装的时候为什么要collect feedback呀？

szjfhb replied:
NO， 这个是 卸载的时候弹的 feedback。 是在安装的时候进行设置。

tonychen2 commented on web-plugin/demo/index.html:
```diff
@@ -0,0 +1,61 @@
+        <button id="clear" style="margin-left: 80px;">Clear Cart</button>
+    </div>
+    <div style="margin: auto; margin-top: 10px; width: 350px; ">
+        <input id="bageText" placeholder="input bage text" />
```
```suggestion
        <input id="badgeText" placeholder="input badge text" />
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+        showItems(carts);
+    }
+    chrome.action.getBadgeText({}, (res) => {
+        $("#bageText").val(res);
```
```suggestion
        $("#badgeText").val(res);
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+});
+
+$("#btnText").click(() => {
+    chrome.action.setBadgeText({ text: $("#bageText").val().trim() });
```
```suggestion
    chrome.action.setBadgeText({ text: $("#badgeText").val().trim() });
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+}
+
+$("#btnColor").click(async () => {
+    resetBageText();
```
```suggestion
    resetBadgeText();
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+}
+
+$("#btnBackRdm").click(async () => {
+    resetBageText();
```
```suggestion
    resetBadgeText();
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+});
+
+$("#btnTxtColor").click(async () => {
+    resetBageText();
```
```suggestion
    resetBadgeText();
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+});
+
+$("#btnTxtRdm").click(async () => {
+    resetBageText();
```
```suggestion
    resetBadgeText();
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+    $("#txtColor").val(rdmColor);
+});
+
+async function resetBageText() {
```
```suggestion
async function resetBadgeText() {
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+});
+
+async function resetBageText() {
+    let uiText = $("#bageText").val();
```
```suggestion
    let uiText = $("#badgeText").val();
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+    let uiEmpty = uiText?.length == 0;
+    if (uiEmpty) {
+        uiText = bageShowText;
+        $("#bageText").val(uiText);
```
```suggestion
        $("#badgeText").val(uiText);
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+
+    if (uiEmpty || uiText != bageShowText) {
+        if (uiEmpty) {
+            $("#bageText").val('(^_*)');
```
```suggestion
            $("#badgeText").val('(^_*)');
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+    let bageShowText = await chrome.action.getBadgeText({});
+    let uiEmpty = uiText?.length == 0;
+    if (uiEmpty) {
+        uiText = bageShowText;
```
```suggestion
        uiText = badgeShowText;
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+
+async function resetBageText() {
+    let uiText = $("#bageText").val();
+    let bageShowText = await chrome.action.getBadgeText({});
```
```suggestion
    let badgeShowText = await chrome.action.getBadgeText({});
```

tonychen2 commented on web-plugin/demo/index.js:
```diff
@@ -0,0 +1,141 @@
+        $("#bageText").val(uiText);
+    }
+
+    if (uiEmpty || uiText != bageShowText) {
```
```suggestion
    if (uiEmpty || uiText != badgeShowText) {
```

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,268 @@
+        status = STATUS.HaveFood;
+    }
+
+    setBadge({
```
有个小问题。有时候badge里显示的和popup里的不一致（比如badge显示没有tiem但是点进去有之前保留的carbon emission）。这个有什么简单的办法解决吗？

szjfhb replied:
以前在计算之前 我没有清理 旧数据。 现在已经进行了清理。 这种情况会不会出现。 不过可能出现的是， 进去是空 或者有东西， 突然变成 有东西， 或者 no - food。 因为 都是后台计算途中， 用户打开了插件。 导致最开始看到的是上一次的数据（这时候应该基本都是空） 要么新加了数据， 要么新数据突然发现没food。
感觉不大好规避， 之前群里讨论过， 可能还是有一个 单独 的 “waiting” page最好。  计算流程一般需要 200~800ms 左右。 我已经增加了 iscalc 的标志。

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,268 @@
+    let status = STATUS.Empty;
+    let itemsCount = items ? items.length : 0;
+
+    if (itemsCount > 0) {
```
不是特别重要，但这里是不是可以稍微简化成 `let status = itemsCount > 0 ? STATUS.HaveFood : STATUS.Empty;` 然后把第150行去掉？

szjfhb replied:
恩恩， 可以打。 ：）

tonychen2 commented on web-plugin/js/common.js:
```diff
@@ -0,0 +1,24 @@
+    ERROR: 'error'
+}
+
+//we need a convert table for like  "ct -> ounce". many convert GHGI support it.
```
```suggestion
//TODO: we need a convert table for like  "ct -> ounce". many convert GHGI support it.
```

szjfhb replied:
yeah, below is the table. :)  that comments can changed to a description.

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+}
+
+function BuildFoodItem(foodName, unit, quantity) {
+    if (unit) {
```
有碰到过没有unit的情况吗？是不是可以直接认为是each，按照151行解决？

szjfhb replied:
没有 unit 的时候 ， 在后面依然是当作 ea处理的。 这个加上去是防止后面处理 unit 报错。 
不过暂时的确没看到过 完全没有 unit 的。。。

szjfhb committed:
Update web-plugin/popup/Outdated.html

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/popup/error.html

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/readme.MD

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/popup/offset.html

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/popup/offset.css

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/demo/index.js

Co-authored-by: Tony <tony.chen@embarktrucks.com>

szjfhb committed:
Update web-plugin/demo/index.js

Co-authored-by: Tony <tony.chen@embarktrucks.com>

Commit:
update  some codes for review.

By Odin

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    if (!notInCartUI && tagName == 'img') {
+        target = target.parentElement;
+        if (target.className == 'RetailerLogo') {
+            delayPrintCart();
```
```suggestion
            waitThenScrapeCart();
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    //view cart click, before function call, the UI attribute is in cart.
+    else if ((!isHomePage && ariaLabel?.includes("view cart")) || ariaLabel?.includes("increment") ||
+        ariaLabel?.includes("decrement") || btnText == 'remove') {
+        delayPrintCart();
```
```suggestion
        waitThenScrapeCart();
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+}
+
+let timerPrint = null;
+async function delayPrintCart() {
```
```suggestion
async function scrapeCart() {
```

tonychen2 replied:
```suggestion
async function waitThenScrapeCart(timeout=500) {
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    timerPrint = setTimeout(printCart, 500);
+}
+
+function printCart() {
```
```suggestion
function scrapeCart(timeoutBeforeRetry=300) {
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+
+    clearTimeout(timerPrint);//cancel the before timer first.
+    console.log(`Go to read  carts, wait 500ms first: ${new Date().toLocaleString()}`);
+    timerPrint = setTimeout(printCart, 500);
```
```suggestion
    timerPrint = setTimeout(scrapeCart, 500);
```

tonychen2 replied:
```suggestion
    timerPrint = setTimeout(scrapeCart, timeout);
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    let cartBody = document.querySelector('div[id="cart-body"]');
+    if (!cartBody) {//here to make sure cartbody is loaded.
+        console.log(`Cart Body not ready, wait 300ms again: ${new Date().toLocaleString()}`);
+        timerPrint = setTimeout(printCart, 300);
```
```suggestion
        timerPrint = setTimeout(scrapeCart, timeoutBeforeRetry);
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    }
+    let cartBody = document.querySelector('div[id="cart-body"]');
+    if (!cartBody) {//here to make sure cartbody is loaded.
+        console.log(`Cart Body not ready, wait 300ms again: ${new Date().toLocaleString()}`);
```
```suggestion
        console.log(`Cart Body not ready, wait ${timeoutBeforeRetry}ms again: ${new Date().toLocaleString()}`);
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    })
+
+    clearTimeout(timerPrint);//cancel the before timer first.
+    console.log(`Go to read  carts, wait 500ms first: ${new Date().toLocaleString()}`);
```
```suggestion
    console.log(`Go to read  carts, wait ${timeout}ms first: ${new Date().toLocaleString()}`);
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    let tagName = target?.tagName?.toLowerCase();
+
+    if (!notInCartUI && tagName == 'img') {
+        target = target.parentElement;
```
```suggestion
```

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+
+    if (!notInCartUI && tagName == 'img') {
+        target = target.parentElement;
+        if (target.className == 'RetailerLogo') {
```
```suggestion
        if (target.parentElement.className == 'RetailerLogo') {
```

tonychen2 replied:
用户点击的只可能是`RetailerLogo` class的child element吗？有没可能直接点到`RetailerLogo`呢？

szjfhb replied:
恩恩,  这里是可能读错的.  要复用下面的那个循环读取逻辑.. 
我觉得把那个个找到 主 class/ID 的操作提成小函数. 恩,

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,157 @@
+    let target = e.target;
+    let tagName = target?.tagName?.toLowerCase();
+
+    if (!notInCartUI && tagName == 'img') {
```
```suggestion
    if (!notInCartUI && tagName == 'img') {
    // Re-scrape if user goes to a different retailer's cart
```

Commit:
Rename functions

szjfhb committed:
1. rename some functions by review.

2. more work on unit. found  lb bag, lb container words.
3. keep the origImpact...
 still need work to scrape the site to found more units....
by Odin

# Conflicts:
#	web-plugin/js/content.js

szjfhb committed:
resolve merge issue.

by Odin

szjfhb committed:
correct origImpact ( total in g) & total Impact (matched in lb).

by Odin

szjfhb committed:
update some for webscraper...

by Odin

tonychen2 commented on web-plugin/js/content.js:
```diff
@@ -0,0 +1,164 @@
+    target = getTargetbyType(target, 'button');
+    let isRetailBtn = target?.querySelector(".RetailerLogo") ? true : false;
+
+    if (isRetailBtn) {
```
这段的逻辑有没可能稍微整理下让可读性强些呢？我看了半天，主要就是四种情况嘛对吧：
- 打开购物车：waitThenScrapeCart()
- 进入别的retailer购物车：waitThenScrapeCart()
- increment：waitThenScrapeCart() （需要wait吗？）
- 添加新商品：提示需要打开购物车
如果放在一个大的if语句里可能会清晰一些，你觉得呢？
-

szjfhb replied:
现在 wait 是 0.5s就激活一次.  不等待会有重复请求的问题, 有时候会导致后台里面乱掉了.
等待是必须的, 比如 你 1s 点了10几次, 从 1 增加到 11,  0.5s 的间隔, timer 会不停取消上一次触发的job.
这样, 0.5内没有新change 触发, 才会真正执行下一步动作.

恩, 这一片读取button 的code, 其实 主要把 retailer 购物车按钮放到外面处理了.  这个其实是 in cart 的行为.
in cart:  增减删 items,  切换 retailer.
out cart:  增减删 items,  提示 过期.

code 一点点加起来, :) 是有点混乱, 我后面看看思考下 怎么写更直观.

tonychen2 commented on web-plugin/popup/offset.html:
```diff
@@ -0,0 +1,49 @@
+
+		<div class="container">
+			<!--insert make cart carbon neutral image-->
+			<div class="child one">
```
```suggestion
			<div class="total-emission">
```

tonychen2 commented on web-plugin/popup/offset.html:
```diff
@@ -0,0 +1,49 @@
+				<span class="DispValue">···</span>
+				<span class="DispExText"> lbs of CO2</span>
+			</div>
+			<div class="child two">
```
```suggestion
			<div class="offset-cost">
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+let cartItems = null;
+const carbonEmission = document.querySelector('div[class="child one"]');
```
```suggestion
const carbonEmission = document.querySelector('div[class="total-emission"]');
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+//TODO: this need move to some other js file...
+function buildItem(impacts) {
+  if (isOffset) {
+    let pEmission = document.querySelector('div[class="child one"] span');
```
```suggestion
    let pEmission = document.querySelector('div[class="total-emission"] span');
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+let cartItems = null;
+const carbonEmission = document.querySelector('div[class="child one"]');
+const carbonCost = document.querySelector('div[class="child two"]');
```
```suggestion
const carbonCost = document.querySelector('div[class="offset-cost"]');
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+    let pEmission = document.querySelector('div[class="child one"] span');
+    pEmission.innerText = impacts.totalImpact.toFixed(1);
+
+    let pCost = document.querySelector('div[class="child two"] span');
```
```suggestion
    let pCost = document.querySelector('div[class="offset-cost"] span');
```

Commit:
Rename to more descriptive HTML classes

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+let cartItems = null;
+const carbonEmission = document.querySelector('div[class="total-emission"]');
+const carbonCost = document.querySelector('div[class="offset-cost"]');
```
这俩好像并没有被用到？

szjfhb replied:
oh, 该换了 data 的添加方式, 这个是没用了.

tonychen2 commented on web-plugin/js/background.js:
```diff
@@ -0,0 +1,254 @@
+    setIsCalcStatus(false);
+}
+
+function setIsCalcStatus(flag) {
```
所以这个isCalc的flag是正在计算的意思对吧？能不能改成叫`calculating`？我一开始还以为是说计算过了/计算好了。

以及建议最好能在同一个scope里成对出现，不然以后容易出bug。比如82行设成了true，但是false是在`handleCartItems()`里

szjfhb replied:
哈哈, 偷懒写到缩写, 前面用的是 isCalcuating.  我统一下.
恩, 关于场景问题, 我统一下. 
现在去掉了 content.js 传来标志, 直接通过 handleCartItems 来设置是否在计算. 这里是真正计算的模块.
之前引入标志好像是想在demo 里引入设置功能, 其实好像没必要了.

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+  timerCalc = setTimeout(() => { checkIsCalc() }, 1000);
+}
+
+async function checkIsCalc() {
```
这个函数是不是其实可以直接并进`loadCarbonImpact()`里去？

szjfhb replied:
no, 这是一个 timer 函数, 用来循环是否需要 loadcarbon...  逻辑上并进去也OK. 但是这样更清晰点.

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+  loadCarbonImpact();
+}
+else {
+  timerCalc = setTimeout(() => { checkIsCalc() }, 1000);
```
这个1000毫秒的目的是？能不能加在comment里？

szjfhb replied:
OK. :) 这个是 和 jenny 讨论的,  她说显示 waiting 那个条目太快就不见了. 所以强制等 1s, 让用户看到变化.

szjfhb replied:
恩, 之前群里也建议过, 我们是否增加一个  完整的 wait 页面, 和 jenny 之前定义一样. 可能还是那样更正规?

我也拿不准了. 是我劝说她改成现在这样的. 哈哈~

szjfhb replied:
其实可以搞些小动画, 比如数字转呀转来表示load... 就不需要其它提示文字之类.  不知道具体那种可行. 因为 大wait 页面, 很快切过来, 也很无用.

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+          buildItem(impacts)
+        }
+        else {
+          this.location.href = 'offset.html';
```
这是在干嘛呀？

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+  }
+}
+
+let leanMoreBtn = document.querySelector('#leanMoreBtn');
```
```suggestion
let learnMoreBtn = document.querySelector('#learnMoreBtn');
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+}
+
+let leanMoreBtn = document.querySelector('#leanMoreBtn');
+if (leanMoreBtn) {
```
```suggestion
if (learnMoreBtn) {
```

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+
+let leanMoreBtn = document.querySelector('#leanMoreBtn');
+if (leanMoreBtn) {
+  leanMoreBtn.addEventListener('click', async () => {
```
```suggestion
  learnMoreBtn.addEventListener('click', async () => {
```

Commit:
Fix typo

tonychen2 commented on web-plugin/js/popup.js:
```diff
@@ -0,0 +1,136 @@
+  });
+}
+
+closeBtn = document.querySelector('.CloseBtn');
```
这俩的区别是？

szjfhb replied:
这个 popup 现在有好几个页面, closebtn类似也有好几个... :)
.Close  是哪个 X 按钮.
Close Btn 是  OutDated 页面, 关闭大按钮.
BackBtn 是 错误页面的 retry 按钮.
 我加下备注.  :)

szjfhb replied:
恩, 对某些button也改名了. :)

szjfhb committed:
changes  for review.

by Odin

szjfhb committed:
remove tofuItem class.

TODO: Do we need a option Page, to let user set the options?
by Odin

szjfhb committed:
1. change icon in empty page.

2. cart read related logic update.
by Odin

szjfhb committed:
1. webscraper capture change.

2. outdated page fix issue.
3. improve content logic for click status check.
by Odin

szjfhb committed:
continue button action check.

by Odin

szjfhb committed:
update some button check logic.

read cart logic seems not easy to do...
by Odin

szjfhb committed:
1. adjust detail related div css format & display format.

2. add animate for emission value field.
by Odin

szjfhb committed:
demo  add set calcuating flag.

by Odin

szjfhb committed:
handle empty to open offset.

by Odin

szjfhb committed:
change to pure CSS animation.

by Odin
